src/datachain/toolkit/split.py

```diff
@@ -1,7 +1,16 @@
+import random
+from typing import Optional
+
 from datachain import C, DataChain
 
+RESOLUTION = 2**31 - 1  # Maximum positive value for a 32-bit signed integer.
```

Comment:
Off the top of my head, distribution doesn't get any "better" by using "the @dreadatour prime"[^1] 😄 for the _modulo_ operation.

We could just use `2**31`; thus, considering only the least significant 31 bits of `sys.rand` and discarding the rest; but it's nicer to use your approach and take into account all the bits.

```suggestion
RESOLUTION = 2**31
```

[^1]: A particular Mersenne prime: `2**32 - 1`